orangecontrib/text/corpus.py

```diff
@@ -100,6 +100,24 @@ def extend_corpus(self, metadata, Y):
         self.X = self.W = np.zeros((len(self), 0))
         Table._init_ids(self)
 
+    def update_attributes(self, X, dictionary):
+        """
+        Adds X to the corpus and updates its domain accordingly.
+        """
+        self.X = np.hstack((self.X, X))
+
+        dict_key_count = len(sorted(dictionary.keys()))
+        features = [dictionary[key] for key in range(dict_key_count)]
+        new_attr = [ContinuousVariable.make(f) for f in features]
+        for existing_attr in self.domain.attributes:
+            new_attr.insert(0, existing_attr)
```

Comment:
This would reverse the order of existing attributes. What about:
```
new_attr = self.domain.attributes
new_attr += [ContinuousVariable.make(f) for f in features]
```